pkgs/by-name/cr/cryptpad/package.nix

```diff
@@ -0,0 +1,81 @@
+{ lib
+, buildNpmPackage
+, fetchFromGitHub
+, makeBinaryWrapper
+, nodejs
+, nixosTests
+}:
+
+let
+  version = "5.6.0";
+in
+buildNpmPackage {
+  inherit version;
+  pname = "cryptpad";
+
+  src = fetchFromGitHub {
+    owner = "cryptpad";
+    repo = "cryptpad";
+    rev = version;
+    hash = "sha256-A3tkXt4eAeg1lobCliGd2PghpkFG5CNYWnquGESx/zo=";
+  };
+
+  npmDepsHash = "sha256-tQUsI5Oz3rkAlxJ1LpolJNqZfKUGKUYSgtuCTzHRcW4=";
+
+  nativeBuildInputs = [
+    makeBinaryWrapper
+  ];
+
+  # cryptpad build tries to write in cache dir
+  makeCacheWritable = true;
+
+  patches = [
+    # fix some config handling. Reported as
+    # https://github.com/cryptpad/cryptpad/pull/1212
+    ./0002-don-t-ignore-config-s-httpAddress.patch
+  ];
+
+  # npm install apparently drops some directories generated during build,
+  # just cp...
+  dontNpmInstall = true;
+  installPhase = ''
+    runHook preInstall
+
+    out_cryptpad="$out/lib/node_modules/cryptpad"
+
+    mkdir -p "$out_cryptpad"
+    cp -r . "$out_cryptpad"
```

Comment:
Did you maybe just miss running `npm run install:components` in `postInstall`? If so, rewriting the whole `installPhase` seems a tad bit excessive.

Reply:
I'll admit I have no idea of how npm works otherwise I wouldn't have resorted to cp here -- I think I ran into troubles where it expected the source tree to be writable or something like that?
I'll give it a new shot as well as `postFixup` next weekend, it's about time I'd clean up the branch a bit to get it ready for merging...

Reply:
Thanks for the pointer to install:components, that installs in the build tree so it still requires some kludging but it's better than the `ln -s ../node_modules www/components`  I had before.

Also fixed to use postInstall